kong/plugins/key-auth/daos.lua

```diff
@@ -11,6 +11,7 @@ return {
       { created_at = typedefs.auto_timestamp_s },
       { consumer = { type = "foreign", reference = "consumers", default = ngx.null, on_delete = "cascade", }, },
       { key = { type = "string", required = false, unique = true, auto = true }, },
+      { secret = { type = "string", required = false, unique = true, auto = true }, },
```

Comment:
the `key` needs to be unique to identify a consumer, but the `secret` has no such constraint?

Reply:
Yes, that's on purpose. A username (secret) needs to be unique, a password/secret does not need to be unique, right? I could even think of some kind of attack where someone create his own consumer (let's say in the developer portal) and tries all secrets and every time he gets a violation he already knows someone else has used this password / secret

Reply:
yes, but the secret currently has `unique = true`, so that should go?

Reply:
You are right - it's false now (in the next commit)